### ISSUE TITLE
How can I register custom language?

### ISSUE THREAD

**Issue by andsilver** [SamVerschueren/ngx-monaco#25]
Hello,

Thanks for this great module. I have one to get help you. I want to register custom language, but how can i do this?
Hope to hear from you soon.

Thanks,

**Comment by ahmadmu** [comment]
Try this:
https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-custom-languages

**Comment by yvz5** [comment]
```
// Register a new language
monaco.languages.register({ id: 'mySpecialLanguage' });
```

what is the equivalent of this then ?

**Comment by simonh1000** [comment]
@ahmadmu  how does that page relate to https://microsoft.github.io/monaco-editor/monarch.html

**Comment by VashLT** [comment]
Is it possible to create a language that extends from another one? For example, I want to register MJML language and use the same syntax highlighting of HTML but change the completion.

**Comment by Memoraike** [comment]
> Is it possible to create a language that extends from another one? For example, I want to register MJML language and use the same syntax highlighting of HTML but change the completion.

Of course, for this you need to get the configurations of the existing language and apply to your.

<details>
<summary>AMD js example</summary>

```js
(() => {
  const CUSTOM_LANG_ID = 'special_language';
  monaco.languages.register({ id: CUSTOM_LANG_ID });
  const js = await new Promise((resolve, reject) => {
    globalThis.require(['vs/basic-languages/javascript/javascript'], resolve, reject);
  });
  monaco.languages.setMonarchTokensProvider(CUSTOM_LANG_ID, js.language);
  monaco.languages.setLanguageConfiguration(CUSTOM_LANG_ID, js.conf);
})
```
</details>
